acquisinator-sw-bootloader/Core/Src/flash.c

```diff
@@ -0,0 +1,720 @@
+#if (BOOT_FLASH_CRYPTO_HOOKS_ENABLE > 0)
+extern blt_bool FlashCryptoDecryptDataHook(blt_addr address, blt_int8u * data, 
+                                           blt_int32u size);
+#endif
+
+
+/****************************************************************************************
+* Function prototypes
+****************************************************************************************/
+static blt_bool  FlashInitBlock(tFlashBlockInfo *block, blt_addr address);
+static tFlashBlockInfo *FlashSwitchBlock(tFlashBlockInfo *block, blt_addr base_addr);
+static blt_bool  FlashAddToBlock(tFlashBlockInfo *block, blt_addr address,
+                                 blt_int8u *data, blt_int32u len);
+static blt_bool  FlashWriteBlock(tFlashBlockInfo *block);
+
+
+/****************************************************************************************
+* Local constant declarations
+****************************************************************************************/
+/** \brief   If desired, it is possible to set BOOT_FLASH_CUSTOM_LAYOUT_ENABLE to > 0
+ *           in blt_conf.h and then implement your own version of the flashLayout[] table
+ *           in a source-file with the name flash_layout.c. This way you customize the
+ *           flash memory size reserved for the bootloader, without having to modify
+ *           the flashLayout[] table in this file directly. This file will then include
+ *           flash_layout.c so there is no need to compile it additionally with your
+ *           project.
+ */
+#if (BOOT_FLASH_CUSTOM_LAYOUT_ENABLE == 0)
+/** \brief   Array wit the layout of the flash memory.
+ *  \details Also controls what part of the flash memory is reserved for the bootloader.
+ *           If the bootloader size changes, the reserved sectors for the bootloader
+ *           might need adjustment to make sure the bootloader doesn't get overwritten.
+ */
+static const tFlashSector flashLayout[] =
+{
+  /* space is reserved for a bootloader configuration with all supported communication
+   * interfaces enabled. when for example only UART is needed, then the space required
+   * for the bootloader can be made a lot smaller here.
+   */
+  /* { 0x08000000, 0x00800 },           flash sector  0 - 2kb (reserved for bootloader)*/
+  /* { 0x08000800, 0x00800 },           flash sector  1 - 2kb (reserved for bootloader)*/
+  /* { 0x08001000, 0x00800 },           flash sector  2 - 2kb (reserved for bootloader)*/
+  /* { 0x08001800, 0x00800 },           flash sector  3 - 2kb (reserved for bootloader)*/
+  // { 0x08002000, 0x00800 },              /* flash sector  4 - 2kb                       */
+  // { 0x08002800, 0x00800 },              /* flash sector  5 - 2kb                       */
+  // { 0x08003000, 0x00800 },              /* flash sector  6 - 2kb                       */
+  // { 0x08003800, 0x00800 },              /* flash sector  7 - 2kb                       */

```

Comment:
Since the application firwmare starts at address `0x08002000` those lines should not be commented out, otherwise the bootloader will allocate that part of memory for itself.
**Remember to add information about the flash layout into our documentation (for more info check openBLT documentation**
```suggestion
  { 0x08002000, 0x00800 },              /* flash sector  4 - 2kb                       */
  { 0x08002800, 0x00800 },              /* flash sector  5 - 2kb                       */
  { 0x08003000, 0x00800 },              /* flash sector  6 - 2kb                       */
  { 0x08003800, 0x00800 },              /* flash sector  7 - 2kb                       */

```